salt/modules/junos.py

```diff
@@ -103,9 +110,16 @@ def facts():
     conn = __proxy__['junos.conn']()
     ret = dict()
     try:
-        facts = conn.facts
-        facts['version_info'] = dict(facts['version_info'])
-        ret['facts'] = facts
+        ret['facts'] = dict(conn.facts)
```

Comment:
Can we create/have a private helper function which have all the code piece which is common to facts and fact_refresh. say for example:

def _facts_serialization(dev_facts):
    facts = dev_facts
    facts['version_info'] = \
        dict(facts['version_info'])
    if 'junos_info' in facts:
        facts['junos_info']['re0']['object'] = \
            dict(facts['junos_info']['re0']['object'])
       ------
    ----
    return facts

then do 
ret['facts'] = _facts_serialization(dict(conn.facts))

makes sense? Else everytime we do some changes, we will have to do it at 2 places.